pkg/functions/function.go

```diff
@@ -87,6 +92,8 @@ type Function struct {
 
 	// Deploy defines the deployment properties for a function
 	Deploy DeploySpec `yaml:"deploy,omitempty"`
+
+	LocalSettings LocalConfig `yaml:"-"`
```

Comment:
I would prefer this just be `.Local`, but that's just a personal preference (I tend to favor brevity when possible; adding qualifiers as necessary :) )